File: sample-java/src/main/java/com/dylanc/loadingstateview/sample/java/animation/FadeAnimatable.java
package com.dylanc.loadingstateview.sample.java.animation;

import android.view.View;

import androidx.annotation.NonNull;

import com.dylanc.loadingstateview.LoadingStateView;
import com.dylanc.loadingstateview.ViewType;

import org.jetbrains.annotations.NotNull;

/**
 * @author Dylan Cai
 */
public class FadeAnimatable implements LoadingStateView.Animatable {

  private static final long DEFAULT_DURATION = 500;
  private final long duration;

  public FadeAnimatable() {
    this(DEFAULT_DURATION);
  }

  public FadeAnimatable(long duration) {
    this.duration = duration;
  }

  @Override
  public void toggleViewsAnimation(@NonNull View showView, @NonNull View hideView, @NonNull Object showViewType, @NonNull Object hideViewType) {
    showView.setAlpha(0);
    showView.setVisibility(View.VISIBLE);
    showView.animate().alpha(1).setDuration(duration);

    if (showViewType == ViewType.LOADING && hideViewType == ViewType.CONTENT) {
      hideView.setVisibility(View.GONE);
    } else {
      hideView.animate().alpha(0).setDuration(duration).withEndAction(() -> {
        hideView.setAlpha(1);
        hideView.setVisibility(View.GONE);
      });
    }
  }
}

File: sample-kotlin/src/main/java/com/dylanc/loadingstateview/sample/kotlin/delegate/FadeAnimatable.kt
package com.dylanc.loadingstateview.sample.kotlin.delegate

import android.view.View
import com.dylanc.loadingstateview.LoadingStateView
import com.dylanc.loadingstateview.ViewType

/**
 * @author Dylan Cai
 */
class FadeAnimatable(private val duration: Long = 600) : LoadingStateView.Animatable {
  override fun toggleViewsAnimation(showView: View, hideView: View, showViewType: Any, hideViewType: Any) {
    if (showViewType === ViewType.LOADING && hideViewType === ViewType.CONTENT) {
      showView.visibility = View.VISIBLE
      hideView.visibility = View.GONE
    } else {
      showView.alpha = 0f
      showView.visibility = View.VISIBLE
      showView.animate().alpha(1f).setDuration(duration)
      hideView.animate().alpha(0f).setDuration(duration).withEndAction {
        hideView.alpha = 1f
        hideView.visibility = View.GONE
      }
    }
  }
}
